### PR TITLE
ECO-3349 Fix Payone dependencies

### PR DESCRIPTION
**Pull request by pyvovarov-s**
- Developer(s): @pyvovarov-s

- Ticket: https://spryker.atlassian.net/browse/ECO-3349

- Academy PR: ACADEMY_URL_HERE

- Release Group: https://release.spryker.com/release-groups/view/2208

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Payone                | minor                 |                       |

#### Release Notes

{skip}

-----------------------------------------

##### Change log

Improvements

Adjusted this module to allow new `spryker/shipment: "^8.0.0"`, `spryker/sales: "^11.0.0"`,  `spryker/oms: "^11.0.0"` and `spryker/payment: "^5.0.0"` versions.

### CODE REVIEW

File: tests/SprykerEcoTest/Zed/Payone/_support/_generated/PayoneZedTesterActions.php [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?php  //[STAMP] 435a9ee73ab6675f928ac55b2f1f853a
+<?php  //[STAMP] 10b9dc2ce48b53ef82167b64a006b184
```
Comment (dereuromark) [inline tests/SprykerEcoTest/Zed/Payone/_support/_generated/PayoneZedTesterActions.php]: these should not be commited.

File: tests/SprykerEcoTest/Zed/Payone/Business/Payment/PaymentLogTest.php [hunk 2]
```diff
@@ -35,7 +35,7 @@ public function testGetPaymentLogs()
         $apiLogs[] = $this->createPayoneApiLog(PayoneApiConstants::REQUEST_TYPE_CAPTURE, PayoneApiConstants::RESPONSE_TYPE_TIMEOUT);
 
         $orderCollection = (new OrderCollectionTransfer())
-            ->addOrders($this->orderTransfer);
+            ->addOrder($this->orderTransfer);
```
Comment (dereuromark) [inline tests/SprykerEcoTest/Zed/Payone/Business/Payment/PaymentLogTest.php]: When did this change? Looks like a BC break.

Comment (pyvovarov-s) [reply tests/SprykerEcoTest/Zed/Payone/Business/Payment/PaymentLogTest.php]: This was released in the previous major. For now, I've only fixed the tests that had been broken in the previous release. "PaymentLogTest.php" - It touches the only test file

File: tests/SprykerEcoTest/Zed/Payone/_support/_generated/PayoneZedTesterActions.php (hunk 1 above)
Comment (pyvovarov-s) [reply tests/SprykerEcoTest/Zed/Payone/_support/_generated/PayoneZedTesterActions.php]: removed

File: src/SprykerEco/Zed/Payone/Business/Payment/PaymentManager.php [hunk 3]
```diff
@@ -78,6 +78,15 @@ class PaymentManager implements PaymentManagerInterface
     public const LOG_TYPE_TRANSACTION_STATUS_LOG = 'SpyPaymentPayoneTransactionStatusLog';
     public const ERROR_ACCESS_DENIED_MESSAGE = 'Access denied';
 
+    /**
+     * @see \Spryker\Shared\Shipment\ShipmentConstants::SHIPMENT_EXPENSE_TYPE | \Spryker\Shared\Shipment\ShipmentConfig::SHIPMENT_EXPENSE_TYPE
```
Comment (tamasnyulas) [inline src/SprykerEco/Zed/Payone/Business/Payment/PaymentManager.php]: separate to 2 lines

Comment (pyvovarov-s) [reply src/SprykerEco/Zed/Payone/Business/Payment/PaymentManager.php]: done

File: src/SprykerEco/Yves/Payone/Dependency/Client/PayoneToShipmentBridge.php [hunk 4]
```diff
@@ -31,6 +31,20 @@ public function __construct($shipmentClient)
      */
     public function getAvailableMethods(QuoteTransfer $quoteTransfer)
     {
+        if (method_exists($this->shipmentClient, 'getAvailableMethodsByShipment') === true) {
+
+            $shipmentMethodsCollectionTransfer = $this->shipmentClient->getAvailableMethodsByShipment($quoteTransfer);
+
+            if ($shipmentMethodsCollectionTransfer->getShipmentMethods()->count() > 1) {
+                throw new \RuntimeException('Split shipping is not supported');
```
Comment (aleksey-kotsuba) [inline src/SprykerEco/Yves/Payone/Dependency/Client/PayoneToShipmentBridge.php]: There should be specific exception.

Comment (pyvovarov-s) [reply src/SprykerEco/Yves/Payone/Dependency/Client/PayoneToShipmentBridge.php]: Already discussed with @dereuromark and @tamasnyulas. Recap left RuntimeException for now

**Pull request merged**